File: leetcode/cpp/58_length_of_last_word.cpp
#include <bits/stdc++.h>
using namespace std;

string reverse(string s)
{
    string ns = "";
    for (char c : s)
    {
        ns = c + ns;
    }
    return ns;
}

string trimLeft(string s)
{
    string ns = "";
    bool isCharFound = false;
    for (char c : s)
    {
        if (!isCharFound && c != ' ')
        {
            isCharFound = true;
        }
        if (isCharFound)
        {
            ns += c;
        }
    }
    return ns;
}

class Solution
{
  public:
    int lengthOfLastWord(string s)
    {
        int l = 0;
        for (char c : trimLeft(reverse(s)))
        {
            cout << c << "\n";
            if (c == ' ')
            {
                break;
            }
            l++;
        }
        return l;
    }
};

int main()
{

    return 0;
};
// github.com/p-nerd (Shihab Mahamud)
// Wednesday, April 17, 2024 | 10:01:13 PM (+6)

File: leetcode/go/58_length_of_last_word.go
package main

func lengthOfLastWord(s string) int {
	x := trimLeft(reverse(s))
	for i, c := range x {
		if c == ' ' {
			return i
		}
	}
	return len(x)
}

func reverse(s string) string {
	r := ""
	for _, c := range s {
		r = string(c) + r
	}
	return r
}

func trimLeft(s string) string {
	p := 0
	for _, c := range s {
		if c == ' ' {
			p++
		} else {
			break
		}
	}
	return s[p:]
}

// github.com/p-nerd (Shihab Mahamud)
// Wednesday, April 17, 2024 | 06:08:37 PM (+6)

